File: testdata/bench/cpp/BenchmarkTest00081.cpp
#include <cstdio>
#include <cstdlib>

int main() {
    printf("Content-Type: text/html\r\n\r\n");
    char *name = getenv("QUERY_STRING");
    printf("<html><body><h1>Hello %s</h1></body></html>", name);
    return 0;
}

File: testdata/bench/rust/BenchmarkTest00081.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Vulnerable: XSS via format! HTML body
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("name").unwrap_or(&String::new()).clone();
    let html = format!("<html><body>Hello, {}!</body></html>", param);
    HttpResponse::Ok().content_type("text/html").body(html)
}
